proxies/venue-graphql-proxy/src/utils/logger.ts

```diff
@@ -0,0 +1,97 @@
+import chalk from 'chalk';
+import AppConfig from '../config/AppConfig';
+import type { Logger, LoggerFunction } from '../types';
+
+function getColor(level: string) {
+  switch (level) {
+    case 'error':
+      return chalk.bold.red;
+    case 'warn':
+      return chalk.bold.yellow;
+    case 'debug':
+      return chalk.cyan;
+    case 'info':
+      return chalk.bold.white;
+    default:
+      return chalk.white;
+  }
+}
+
+function formatMessage(
+  namespace: string,
+  level: string,
+  message?: Parameters<LoggerFunction>[0],
+  ...optionalParameters: Parameters<LoggerFunction>[1][]
+) {
+  const color = getColor(level);
+  const tags = [level, namespace];
+  const renderedTag = tags.map((tag) => `[${tag}]`).join(' ');
+  const messageData = {
+    level,
+    timestamp: new Date().toJSON(),
+    message: [`${renderedTag} - `, message, ...optionalParameters].join(' '),
+  };
+
+  return [
+    new Date(messageData.timestamp).toLocaleTimeString(),
+    `${color(renderedTag)} -`,
+    message,
+  ].join(' ');
+}
+
+function createLogger(namespace: string, debug = false): Logger {
+  return {
+    debug: (
+      message?: Parameters<LoggerFunction>[0],
+      ...optionalParameters: Parameters<LoggerFunction>[1][]
+    ) =>
+      debug &&
+      // eslint-disable-next-line no-console
+      console.debug(
+        formatMessage(namespace, 'debug', message, ...optionalParameters)
+      ),
+    info: (
+      message?: Parameters<LoggerFunction>[0],
+      ...optionalParameters: Parameters<LoggerFunction>[1][]
+    ) =>
+      // eslint-disable-next-line no-console
+      console.info(
+        formatMessage(namespace, 'info', message, ...optionalParameters)
+      ),
+    warn: (
+      message?: Parameters<LoggerFunction>[0],
+      ...optionalParameters: Parameters<LoggerFunction>[1][]
+    ) =>
+      // eslint-disable-next-line no-console
+      console.warn(
+        formatMessage(namespace, 'warn', message, ...optionalParameters)
+      ),
+    error: (
+      message?: Parameters<LoggerFunction>[0],
+      ...optionalParameters: Parameters<LoggerFunction>[1][]
+    ) =>
+      // eslint-disable-next-line no-console
+      console.error(
+        formatMessage(namespace, 'error', message, ...optionalParameters)
+      ),
+  };
+}
```

Comment:
Could the logger from the components-package be used instead? It seems to be an exact match. `components/src/loggers/logger.ts`

Reply:
I just copy-pasted it because it's very odd that server rpxy would depend on components package. I think we could have a separate package for more common stuff for isomorfic code that works everywhere, perhaps the name could be just "common" or something. This logger could be there first and later something else like types, constants, util functions etc.

Reply:
Also `proxies/events-graphql-proxy` uses winston logger and we could use that here as well. Now `proxies/venue-graphql-proxy` code is copy-pasted and doesnt fit so well to `packages/graphql-proxy-server`. In the best situation all the venue code could have one big context object that startServer command takes in. That's how `proxies/events-graphql-proxy` does